### PR TITLE
DEV2-4433 chat api usage

### PR DESCRIPTION
**Pull request by dimacodota**
DEV2-4433 chat api usage

highlights:
 the chat api is now a class that exposes 
 - the handle Message api 
 - handles the chat incoming requests
  
this class is inited differently from the saas extension and the enterprise extension

### CODE REVIEW

File: src/tabnineChatWidget/ChatApi.ts [hunk 1]
```diff
@@ -94,162 +90,167 @@ type WorkspaceFolders = {
...
-    context.subscriptions.push(
-      vscode.commands.registerCommand(
-        GET_CHAT_STATE_COMMAND,
+type APIConfig = {
+  serverUrl?: string | undefined;
+  isSelfHosted: boolean;
+  isTelemetryEnabled: boolean;
+};
+
+export class ChatApi {
```
Comment (dimacodota) [inline src/tabnineChatWidget/ChatApi.ts:99]: this is  now a class that is inited [here](https://github.com/codota/tabnine-vscode/pull/1397/files#diff-c33546e0b57fc3ecb21320676107561d142900bbc9aa67ba0b18808cfd0b4e9fR120)
and [here](https://github.com/codota/tabnine-vscode/pull/1397/files#diff-04bba6a35cad1c794cbbe677678a51de13441b7a6ee8592b7b50be1f05c6f626R156)

File: src/tabnineChatWidget/ChatApi.ts [hunk 2]
```diff
@@ -94,162 +90,167 @@ type WorkspaceFolders = {
...
-      onInit();
-      return Promise.resolve({
-        ide: "vscode",
-        isDarkTheme: [
-          ColorThemeKind.HighContrast,
-          ColorThemeKind.Dark,
-        ].includes(vscode.window.activeColorTheme.kind),
-        isTelemetryEnabled: isCapabilityEnabled(Capability.ALPHA_CAPABILITY),
-        serverUrl,
-        isSelfHosted: context.workspaceState.get(
```
Comment (dimacodota) [inline src/tabnineChatWidget/ChatApi.ts:125]: @yonidavidson this was changed to be injected from the ctor, and not managed as a flag

**Review by yonip23 — APPROVED** [review]
lgtm, nice change 👍🏻

File: src/tabnineChatWidget/ChatApi.ts (hunk 2 above)
Comment (dimacodota) [reply src/tabnineChatWidget/ChatApi.ts:125]: meant @yonip23  and not @yonidavidson  🤦‍♂️

**Review by yairco1990 — APPROVED** [review]
lgtm

**Pull request merged**